Create corplots-PR-v2

AntoniaChroni opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?

This is a script written to create corplots for patient cases with multiple biospecimen samples and matched longitudinal samples.

#### What was your approach?



#### What GitHub issue does your pull request address?



### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?

Please review for script logic.

#### Is there anything that you want to discuss further?

The input file `maf_autopsy.tsv` is generated from the tmb-vaf-preprocess data (2/N) [#16](https://github.com/d3b-center/pbta-tumor-evolution/pull/16) and is placed in `../../scratch`. Please run that script first to ensure that you have all files necessary.

This is a PR based on the previous https://github.com/d3b-center/pbta-tumor-evolution/pull/20 that was closed by accident.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes.

### Results

#### What types of results are included (e.g., table, figure)?

There are 139 plots generated based on every combination between `Kids_First_Biospecimen_ID/tumor_descriptor/Kids_First_Participant_ID` and are placed in `../../plots`.
Please check if that is the number of plots generated. For easier visualization and going through the results, please see the html report.

#### What is your summary of the results?

There are 29 autopsy samples out of the total 119 patient cases (with genomic assays) with maf information. There are 117 (out of the 118) patient samples with TMB information. There are 44 biospecimen samples missing from both TMB and VAF files. 

- Genes shown in the plots are based on the oncoprint goi list from OpenPedCan.
- Multiple plots are generated based on the number of biospecimen samples per tumor descriptor.
- Be aware of differences between plots with the same biospecimen samples. These might show in one as tumor descriptor-specific and in another as common.
- Multiple biospecimen samples/tumor descriptor capture a variety of heterogeneity. We should consider including all biospecimen samples/tumor descriptor and merge that information into one.
- Deceased samples have higher VAFs overall compared to their counterparts in other timepoints.
- Biospecimen samples are from different tumor locations. We will obtain that information after the Nautilus harmonization (primary_site column in histologies). If so, that would allow us to perform spatial heterogeneity analysis ([ticket #19](https://github.com/d3b-center/pbta-tumor-evolution/issues/19)).
- "PT_3CHB9PK5", "PT_6N825561": These samples are hyper-mutant compared to the rest of the samples (VAF corplot).

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

#### Documentation Checklist

<!-- Please review https://github.com/d3b-center/pbta-tumor-evolution#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [x] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [x] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

Commit:
Upload all files to create corplots

Commit:
Run create-corplots

Commit:
Clean code based on Lint

Commit:
Change color order for timepoints

Commit:
Upload plots with new colors

Commit:
Update

Commit:
results from corplots

Commit:
run corplots 2

Commit:
Edit summary in the README file

Commit:
Add Results section in README file

Commit:
Add Results section

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Adjust text size in plot.title

Commit:
Run Rmd in Docker and update plots and html

Commit:
Update README.md

Commit:
Update Results section in the README file

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Replace colors in the function

Commit:
Add color_palette tsv files

Commit:
Add timepoint_color_palette file and run in Docker

Commit:
Insert Roxygen skeleton into the function

Commit:
Edit title size in ggplot

Commit:
Run updated script via Docker

Commit:
Remove palettes - this will be added after merging PR#21

AntoniaChroni committed:
Update analyses/tmb-vaf-longitudinal/util/function-create-corplot.R

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/tmb-vaf-longitudinal/util/function-create-corplot.R

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

Commit:
Add temp palette

Commit:
Bring back the code for renaming colnames(timepoint_df)

Commit:
Bring back code for renaming colnames(deceased_df)

Commit:
Use timepoints_other_plot to create label for timepoint

Commit:
Fix typo in function for creating df for timepoints_deceased_plot

Commit:
Remove temp palettes

Commit:
Make corplot reproducible by adding set.seed

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Update README file about the documentation for the timepoint models

Commit:
Edit dir where theme.R lives

Commit:
Add cancer_group information in the plot as subtitle in the function

Commit:
Add max.overlaps=Inf to the function to avoid overlaps in the plots

Commit:
Update plots and html with cancer_group

Commit:
Merge branch 'create-pptc-pdx-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Rename palette_df

Commit:
Rename palette_df in Rmd

Commit:
Run Rmd

Commit:
Update 02-create-corplots.nb.html

Commit:
Merge branch 'main' into tmb-vaf-longitudinal-create-corplots

Commit:
Run Rmd under Docker to investigate PT_3CHB9PK5 plot

Commit:
Merge branch 'main' into tmb-vaf-longitudinal-create-corplots

Commit:
Edit #patients with TMB info in README

AntoniaChroni commented:
@naqvia this is related to https://github.com/d3b-center/pbta-tumor-evolution/pull/20. It was closed by accident. See below my answer to your comment:

This plot is actually generated and is shown in the html report. I am not sure why this was previously not saved properly. I run the script again under Docker and it is there. This is actually the most large file of all (921 kb) because there is a high number of mutations for this sample. Can you please run this again and let me know if you still see a zero generated file and if the plot is in the html or not?

jharenza reviewed (commented):
Going through some of the plots and making some notes:
- for PT_1H2REHT2, can you check Progressive_BS_7M7JNG00 and Deceased_BS_96S0VQBN - it seems they both have a label for NF1_p.Q1822*, but they are not colored grey, but instead look to be specific. Looks right on the plot between Progressive_BS_79SYEHY3 and Deceased_BS_96S0VQBN.
- PT_3KM9W8S8 - looks like Deceased_BS_EQQZ90JQ and Deceased_BS_P8FXRTZ2 are low tumor purity and Deceased_BS_MS8TBCWJ is either a bad sample OR this is starting to get at tumor dissemmination, we should investigate and look at which regions exactly these are coming from using [this](https://chop365-my.sharepoint.com/:x:/g/personal/colemann2_chop_edu/EacM_OjGqY1OrW-HpBBu40QBjCM3QeCtf53ootHlfHSy0A?e=okeaVe).
- PT_6N825561 - looks like Deceased_BS_0YWFFRYV low quality or low tumor
- PT_8GN3TQRM ALK mutation - is this an IHG-ALK subtype?
- PT_23NZGSRJ might also be a good one to look at tumor heterogeneity
- PT_DVXE38EX we should look at dx vs progression as well - is the deceased low purity? not seeing NRAS? was this pt treated with a ras inhibitor and NRAS went away?
- Check PT_5NS35B66 TP53_p.R248W showing unique for each Deceased_BS_F0JN3TFM and Diagnosis_BS_SVKMG97S instead of common
- PT_GTHZF21E and PT_KTRJ8TFY also good for heterogeneity. PT_KTRJ8TFY has multiple TP53 and only in certain deceased samples, maybe region-specific
- Interested in putting together PT_KZ56XHJT longitudinal story with TP53 and SMARCB1
- PT_WYXTEG3E Progressive_BS_G9B420R9 probably low tumor purity - CTNNB1 should be there
- PT_Z4BF2NSB also interesting to look at tumor heterogeneity

I didn't get a chance to rerun yet, but wanted to send this back esp to review the few where the mutations aren't showing up in common.

naqvia reviewed (approved):
Not sure why that one plot was empty before (docker shouldn't have affected that), but I have re-ran the script and it runs as expected, and I con confirm all 139 plots and  resulting txt files.

AntoniaChroni commented:
The number of plots should be 140 and they are not txt files generated in this script. @naqvia

AntoniaChroni commented:
When I click on `get info `on the folder on my computer, it says that contains 140 items. When I select the items in the folder, it says 139. So, please disregard the comment on the #plots. @naqvia

Commit:
Increase xintercept and yintercept and add documentation

Commit:
Move gene labels and add segments

Commit:
Remove segments

because they show for all points instead of only for the labels of the common gene mutations

Commit:
Add file path to results_dir and save maf_join table per pair

Commit:
Remove subtitle

Commit:
Edit documentation about set.seed

Commit:
Add code for Save plots by cancer_group

Commit:
Add subtitle

Commit:
Edit code for Save results by cancer_group

Commit:
Run code and save plots per cancer group

Commit:
Remove old plots

Commit:
Add new plots

Commit:
Edit code on creating df for cancer_group and sample

Commit:
Add cg_sum to identify samples with different diagnosis across timepoints

Commit:
Create df for cancer_group and samples

Commit:
Code for single vs multiple diagnosis samples deactivated

Commit:
Add cg_results_dir

Commit:
Remove gap

Commit:
Save df in the function and add kids_id in df

Commit:
Save df in the function adn add kids_id in df

Commit:
Fix minot typos

Commit:
Fix minor typos

Commit:
Merge remote-tracking branch 'origin/tmb-vaf-longitudinal-create-corplots' into tmb-vaf-longitudinal-create-corplots

Commit:
Add cg_sum and edit code to save plots and results based on this

Commit:
Remove code with deactivated if statement

Commit:
Remove old plots

Commit:
Clean code

Commit:
Revert "Clean code"

This reverts commit 183003124f3b322735d5465c55b8ba1387720b40.

Commit:
Revert "Replace x with names in the loop for cg_sum"

This reverts commit be8bc69b45743fcb49ba554d52798ee59a8a2ad3.

Commit:
Replace x with names in the loop for cg_sum

Commit:
Clean code

Commit:
Add code to sort cg_names

Commit:
Add code chunk for Print df per pair in a list

Commit:
Add   library(reshape2) to reshape df before save

Commit:
Edit order of print plots that caused error

Commit:
Fix extra idents

Commit:
Upload plots and html

Commit:
Rename tsv file with all df with common genes

Commit:
Delete 04-tmb-unpaired-samples-plot.Rmd

Commit:
plots and results run in ec2

Commit:
Remove not necessary files after run to ec2

Commit:
Remove not necessary files - there was conflict

Commit:
Remove print(m)

Commit:
New plots and stop run

Commit:
Add new plots

Commit:
Run in ec2, add new plots, results, and html

AntoniaChroni commented:
@rjcorb The following is to address the question asked about intercepts in https://github.com/d3b-center/pbta-tumor-evolution/pull/20.

''The intercepts here were used to differentiate clonal from subclonal gene mutations. So, it is okay if the `common mutations` cluster close to a specific timepoint.''

However, after discussing with @jharenza, we decided to also address the following:

1. Increase the intercepts to reflect literature values. See 8f8c7942d9eeb83d7d19b04fcdb91c929fb75dfa
2. Decrease circle size in corplots and move gene labels. See 1a8d347daf814568bfc732de7baf079263936fc5
3. Remove the subtitle from plots: @jharenza  I ended up keeping this because it makes it easier to assess the plots in the html report. If you think that it doesn't add value, I can remove it.
4. Identify and take into account cases with multiple diagnosis when plotting. See 26c4bc6b19871bdce61d126797b547ed8bc51222 and 577bab8d65d962f2fbd1e209b06f39bb539b5fde, respectively. In the current dataset, there is one `PT_1H2REHT2` patient case with a secondary diagnosis in one of the progressive samples, but for the deceased sample is all primary diagnosis. I found it simpler to code this by creating `cg_sum` column to summarize the cancer groups. But let me know if there is another way to go around this. Plots are now saved by `cg_sum`.
5. Save table that will contain the information about genes and genes in common per pair: 
     - Added `create_corplot_melt` function to reshape df to homogenize the table format across samples to make it easier to save 
     - Made a list of df and saved all in one table.  See 876a6f8ff79254b3426f5d3dddd427a6a6144144
6. I recommend the reviewers to use the ec2 instance to run the shell script to reduce computational time (as suggested by @naqvia).

Commit:
Merge branch 'main' into tmb-vaf-longitudinal-create-corplots

Commit:
Merge branch 'main' into tmb-vaf-longitudinal-create-corplots

rjcorb reviewed (commented):
This is looking good! I really like the look of the correlation plots. I did notice that, in some cases, the labels don't seem to be aligning with points on the plot. For example, in `PT_MDWPRDBT-Recurrence_BS_00TRPEQX-vs-Deceased_BS_1M63B97V-vaf-corplot.pdf`, it is not clear which variants the labels correspond to. The colors also seem to not match with positions on the plot. Like in that example, shouldn't blue text point to blue points?

AntoniaChroni commented:
@rjcorb The labels can't be positioned exactly on the points on the plot. 

I tried to draw connecting lines that would label specific points on a plot, but with no lack. I was ending up having them in all the points. :(

naqvia commented:
> @rjcorb The labels can't be positioned exactly on the points on the plot.
> 
> I tried to draw connecting lines that would label specific points on a plot, but with no lack. I was ending up having them in all the points. :(

Have you tried `ggrepel`? It automatically would draw a line to the point with the label if there too close together. Here are some examples: https://ggrepel.slowkow.com/articles/examples.html

AntoniaChroni commented:
> > @rjcorb The labels can't be positioned exactly on the points on the plot.
> > I tried to draw connecting lines that would label specific points on a plot, but with no lack. I was ending up having them in all the points. :(
> 
> Have you tried `ggrepel`? It automatically would draw a line to the point with the label if there too close together. Here are some examples: https://ggrepel.slowkow.com/articles/examples.html

Yes, I tried multiple times and it was drawing all lines instead.

rjcorb commented:
@AntoniaChroni ah I see. I will approve since this only seems to be an issue for a few plots. But since some plots have many overlapping labels that make them difficult to read, maybe you could filter points that get labeled based on a VAF threshold?
